server/src/main/java/org/cloudfoundry/identity/uaa/oauth/KeyInfo.java

```diff
@@ -161,11 +175,12 @@ class RsaKeyInfo extends KeyInfo {
     private Signer signer;
     private SignatureVerifier verifier;
     private String verifierKey;
+    private String verifierCertificate;
 
     public RsaKeyInfo(String keyId, String signingKey, String keyUrl) {
-        this(keyId, signingKey, keyUrl, null);
+        this(keyId, signingKey, keyUrl, null, null);
```

Comment:
with https://github.com/cloudfoundry/uaa/pull/2208 we can refactor the other null, however from experieces I have Optional does not really solve it because now you know xxx != null is needed or ObjectsUtils.isNull / isEmpty.
.... with Optinal you need to be carefully with the .get()
So I have not see less Nullpointer expections only moving to Optinal, but we can do the refactor.

Reply:
Use internally Optional and assign it with Optional.ofNullable(signingCert);